pkg/core/blockchain.go

```diff
@@ -478,15 +452,21 @@ func (bc *Blockchain) headerListLen() (n int) {
 
 // GetTransaction returns a TX and its height by the given hash.
 func (bc *Blockchain) GetTransaction(hash util.Uint256) (*transaction.Transaction, uint32, error) {
-	if tx, height, ok := bc.blockCache.GetTransaction(hash); ok {
-		return tx, height, nil
-	}
 	if tx, ok := bc.memPool.TryGetValue(hash); ok {
 		return tx, 0, nil // the height is not actually defined for memPool transaction. Not sure if zero is a good number in this case.
 	}
+	tx, height, err := getTransactionFromStore(bc.memStore, hash)
```

Comment:
curios: why we check tx from memstore after mempool now?

Reply:
Because it was always checked for in the `Store` and `memPool`/`memStore` are different mechanisms for different purposes. `memPool` is kinda staging area for transactions (like the ones sent via `sendrawtransaction` RPC call), I'm not sure we'll keep it long-term (I don't feel like it belongs to `Blockchain`) , but it might be useful for consensus nodes.

Reply:
got it!